test/e2e/network/service.go

```diff
@@ -4541,14 +4616,14 @@ func checkReachabilityFromPod(ctx context.Context, expectToBeReachable bool, tim
 
 // checkServiceReachabilityFromExecPod creates a dedicated client pod, executes into it,
 // and checks reachability to the specified target host and port.
-func checkServiceReachabilityFromExecPod(ctx context.Context, client clientset.Interface, namespace, name, clusterIP string, port int32) {
+func checkServiceReachabilityFromExecPod(ctx context.Context, client clientset.Interface, namespace, name, clusterIP string, port int32) string {
 	// We avoid relying on DNS lookup with the service name here because
 	// we only want to test whether the named port is accessible from the service.
 	serverHost := net.JoinHostPort(clusterIP, strconv.Itoa(int(port)))
 	ginkgo.By("creating a dedicated client to send request to the http server " + serverHost)
 	execPod := e2epod.CreateExecPodOrFail(ctx, client, namespace, "execpod-", nil)
 	execPodName := execPod.Name
-	cmd := fmt.Sprintf("curl -q -s --connect-timeout 2 http://%s/", serverHost)
+	cmd := fmt.Sprintf("curl -q -s --connect-timeout 2 http://%s/hostname", serverHost)
```

Comment:
Note for myself: Maybe `curl` shouldn't be used for conformance test while using `net/http` instead?

Reply:
eh, we use lots of `curl` in the tests.

If you want to use `net/http` from within `e2e.test` itself, then you need to use a service of type `LoadBalancer`, since `e2e.test` runs from outside the cluster.